src/compiler/checker.ts

```diff
@@ -48243,12 +48283,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isExpandoFunctionDeclaration(node: Declaration): boolean {
-        const declaration = getParseTreeNode(node, isFunctionDeclaration);
+        const declaration = getParseTreeNode(node, (n): n is FunctionDeclaration | VariableDeclaration => isFunctionDeclaration(n) || isVariableDeclaration(n));
         if (!declaration) {
             return false;
         }
-        const symbol = getSymbolOfDeclaration(declaration);
-        if (!symbol || !(symbol.flags & SymbolFlags.Function)) {
+        let symbol: Symbol;
+        if (isVariableDeclaration(declaration)) {
+            if (declaration.type || !isVarConstLike(declaration)) {

```

Comment:
This is only true in `TS` (where only declared-with-function-syntax things are expando) - in JS,
```ts
let a = function() {};
a.bar = 1;
```
is both valid, and expando.

cc @sandersn for eyes on this part of the change, since you probably have the most context on what is and is not "expando".

Reply:
In JS, besides arrows, function expressions and declarations, expando includes classes and empty object literals. I'll take a look at this part.

Reply:
Oh, also *non*-empty object literals, when assigned to class prototypes, so

```
C.prototype = {
  method() { // ...
  }
}
C.prototype.othermethod = function() { }
```

I'm not sure that makes sense here, but I think it does.

Reply:
Changed to use `getDeclaredExpandoInitializer` and to return true for JS functions as well.